packages/edit-site/src/components/sidebar/style.scss

```diff
@@ -3,6 +3,28 @@
 	overflow-y: auto;
 }
 
+@keyframes slide-from-right {
+	from {
+		transform: translateX(50px);
+		opacity: 0;
+	}
+	to {
+		transform: none;
+		opacity: 1;
+	}
+}
+
+@keyframes slide-from-left {
+	from {
+		transform: translateX(-50px);
+		opacity: 0;
+	}
+	to {
+		transform: none;
+		opacity: 1;
+	}
+}
+
```

Comment:
We kinda have this for `Navigator` already, although it's using Emotion. Would be nice to duplicate it in some point in the future:

https://github.com/WordPress/gutenberg/blob/2114141f29e3512fd7ba5b538c11eb05b35e3e09/packages/components/src/navigator/styles.ts#L16-L62

Reply:
Yes, and maybe it would be a good idea to migrate the `Navigator` animation away from Emotion. The plain SCSS is simpler, and it actually suprised me seeing that I got the RTL support automatically.

Reply:
Yes - ideally, we'll completely migrate from Emotion, this is something we've been discussing a lot with @mirka and @DaniGuardiola recently

Reply:
> Important note, keyframe names are global, so it'll need a very specific name

This is one advantage of the Emotion implementation, it gives the animations a unique name with hex suffix, like `animation-1f5bucw`. I think we should be able to achieve the same thing with some CSS module feature that marks CSS selectors as local.

Reply:
I'm not sure if I like the `--direction` approach 🙂 Seems to me that we're at a point where the animations are not yet complex enough to warrant such abstractions, and just writing the same thing twice is more straightforward.

Reply:
> I'm not sure if I like the --direction approach 🙂 Seems to me that we're at a point where the animations are not yet complex enough to warrant such abstractions, and just writing the same thing twice is more straightforward.

I guess it's a matter of taste. I wouldn't personally even call this an abstraction, just a different implementation, it's all good tho!

Btw, just realized that you could also omit the `to` part as well IIRC, and the entire thing would become something like this:

```css
@keyframes slide {
	from {
		transform: translateX(calc(var(--direction, 1) * 50px));
		opacity: 0;
	}
}

&:dir(rtl) {
    --direction: -1;
}

@media not (prefers-reduced-motion: reduce) {
    animation-duration: 0.14s;
	animation-timing-function: ease-in-out;
}
```

Reply:
Nice! I can omit the `to` part because I'm animating to the `initial` values of `transform` and `opacity`?

The `@media not` syntax is nice, makes the code simpler -- instead of enabling the animation in general and the disabling it in particular, I just enable it once :+1:

The `dir(rtl)` part looks like a misunderstanding: we really want two kinds of sliding, one from the left, one from the right. It's just that left and right mean something different in RTL.

I'm wondering if `rtlcss` will also change `50px` to `-50px` when it sees the `calc(var(--direction, 1) * 50px)` expression. Will try it out!

Reply:
> Nice! I can omit the to part because I'm animating to the initial values of transform and opacity?

Not 100% sure, but that's what I meant yes!

> The dir(rtl) part looks like a misunderstanding: we really want two kinds of sliding, one from the left, one from the right. It's just that left and right mean something different in RTL.

Ah, you're right, I did misunderstand that! The `dir` pseudoselector might still be helpful to invert the direction in each variant. A note on this, if this sliding animation is supposed to be RTL-aware, consider changing the naming to "start" and "end" instead of "left" and "right", to align with the standard naming (e.g. `margin-start/end` CSS properties).

> I'm wondering if rtlcss will also change 50px to -50px when it sees the calc(var(--direction, 1) * 50px) expression. Will try it out!

Not familiar with `rtlcss`, I'll try to look into it at some point - I might be wrong but if this is an old library I suspect it can probably be replaced by modern CSS features ✨